### ISSUE TITLE
Request for Native Android Apps Support for LightGBM: Android SDK or JNI Integration

### ISSUE THREAD

**Issue by Tejas-Kanani** [microsoft/LightGBM#6592]
## Summary
I would like to request the addition of direct Android support for LightGBM. Currently, LightGBM is primarily used on desktop and server environments, but extending support to Android could significantly benefit mobile developers who need efficient gradient boosting for machine learning tasks on mobile devices.

## Motivation

Although LightGBM is not natively supported for mobile environments, such as Android, it is currently commonly utilised on PCs and servers. LightGBM may become more adaptable and available for mobile developers who need advanced machine learning features on-device by adding support for Android. Better performance and offline functioning for mobile apps that use gradient-boosting models may result from this.

## Description

Android SDK: Provide a native Android SDK for LightGBM so that programmers may incorporate and utilise LightGBM in their Android apps. On the mobile device, this SDK would take care of model training and prediction.

Thank you for considering this request. Native Android support would greatly enhance the usability of LightGBM in mobile applications.

**Comment by gopalvaghelatek** [comment]
Also looking for the same, please update if any solutions are available.

Thank you

**Comment by jameslamb** [comment]
Thanks very much for using LightGBM.

As far as I know, none of the maintainers here are familiar with developing mobile apps. I think it's unlikely that you'll see an Android SDK shipped directly from this repo. But if you're interested in seeing that happen, the best way you can help is to provide more details and to work with us.

Questions I have:

* why can you not just compile `lib_lightgbm.so` (dynamic library) or `lib_lightgbm.a` (static library) and ship it as part of your app, calling into the library through its C API?
* are there other machine learning frameworks that provide an Android SDK, whose API you'd want LightGBM to emulate? Can you share some links to them?
* how would we test this in CI? Are you willing to help with that?

The last time someone came asking for this, I asked some questions similar to this and the original poster never responded: https://github.com/microsoft/LightGBM/issues/6200#issuecomment-1890301973

Here's another example where someone opened a report asking for help with using LightGBM in an Android app, but then disappeared and never responded to our questions: https://github.com/microsoft/LightGBM/issues/6343#issuecomment-2073867256

Other links you may want to follow:

* https://github.com/dmlc/xgboost/issues/10656
* https://github.com/dmlc/xgboost/issues/7342
* https://github.com/catboost/catboost/issues/2496

**Comment by jameslamb** [comment]
Since this has now been asked several times, I've added it to our list of feature requests here: #2302.

We keep feature request issues tracked there, and close them here to keep the issue tracker focused on bugs, questions, and in-progress work. So I'll close this for now, but if you're interested in helping contribute something like this please do comment and we'd be happy to talk more about it!

**Comment by jameslamb** [comment]
Adding these docs: https://docs.python.org/3/using/android.html#building-a-python-package-for-android

especially:

> **6.2. Building a Python package for Android*
>
> _Python packages can be built for Android as wheels and released on PyPI. The recommended tool for doing this is [cibuildwheel](https://cibuildwheel.pypa.io/en/stable/platforms/#android), which automates all the details of setting up a cross-compilation environment, building the wheel, and testing it on an emulator._

For the Python package (which I guess based on #6951 some people want to use on Android), we might be able to build Android wheels.
